bst_plugin/process_nst_dOD_SSR.m

```diff
@@ -0,0 +1,299 @@
+    else
+    [nirs_dOD, channel_dOD_def] = Compute(fnirs, baseline_method, fchannel_def, do_SuperficalRegression);
+    end
+    
+    [final_nirs, ChannelMat] = process_nst_mbll('concatenate_data',nirs_dOD, channel_dOD_def, nirs_other, channel_def_other);
+    
+    % Create new condition because channel definition is different from original one
+     cond_name = sInputs.Condition;
+     if strcmp(cond_name(1:4), '@raw')
+        cond_name = cond_name(5:end);
+     end
+     if do_SuperficalRegression
+         iStudy = db_add_condition(sInputs.SubjectName, [cond_name, '_dOD_SSR']);
+     else
+         iStudy = db_add_condition(sInputs.SubjectName, [cond_name, '_dOD']);
+     end
+     sStudy = bst_get('Study', iStudy);
+%     
+%     % Save channel definition
+     [tmp, iChannelStudy] = bst_get('ChannelForStudy', iStudy);
+     db_set_channel(iChannelStudy, ChannelMat, 0, 0);
+        
+    % Save time-series data
+%     final_nirs = sDataIn.F;
+%     final_nirs(to_keep, :) = final_nirs';
+    sDataOut = db_template('data');
+    sDataOut.F            = final_nirs';
+    sDataOut.Comment      = 'NIRS dOD';
+    %sDataOut.ChannelFlag  = sDataIn.ChannelFlag;
+    sDataOut.ChannelFlag  = ones(size(final_nirs, 2), 1);
+    sDataOut.Time         = sDataIn.Time;
+    sDataOut.DataType     = 'recordings'; 
+    sDataOut.nAvg         = 1;
+    sDataOut.Events       = events;
+    sDataOut.DisplayUnits = 'delta OD';
+
+    % Generate a new file name in the same folder
+    OutputFile = bst_process('GetNewFilename', bst_fileparts(sStudy.FileName), 'data_hb');
+    sDataOut.FileName = file_short(OutputFile);
+    bst_save(OutputFile, sDataOut, 'v7');
+    % Register in database
+    db_add_data(iStudy, OutputFile, sDataOut);
+end
+
+
+function [nirs_dOD, channel_dOD_def] = Compute(nirs_sig, normalize_method, channel_def, do_SuperficalRegression, supf_chann)
+%% Normalize given nirs signal
+% Args:
+%    - nirs_sig: matrix of double, size:  nb_samples x nb_channels
+%        NIRS signal to normalize
+%   [- method]: str, choices are: 'mean' and 'median', default is 'mean'
+%        Normalization method.
+%        * 'mean': divide given nirs signal by its mean, for each wavelength
+%        * 'median': divide given nirs signal by its median, for each wavelength
+% 
+% Output: matrix of double, size: nb_samples x nb_channels
+%    Normalized NIRS signal.
+
+switch normalize_method
+    case 'mean'
+        od_ref = mean(nirs_sig, 1);
+    case 'median'
+        od_ref = median(nirs_sig, 1);
+end
+
+[nirs_psig, pair_names, pair_loc, pair_indexes] = process_nst_mbll('group_paired_channels',nirs_sig, channel_def);
+pair_distances = process_nst_mbll('cpt_distances',channel_def.Channel, pair_indexes) .* 100; %convert to cm
+nb_samples = size(nirs_sig, 1);
+nb_pairs = length(pair_names);
+dOD = zeros(nb_pairs, 2, nb_samples);
+
+if do_SuperficalRegression
```

Comment:
You could remove the argument `do_SuperficalRegression` and rely on the content of `supf_chann`

`if do_SuperficalRegression` ->` if ~isempty(supf_chann)`